modules/access_affinitygroup/src/Plugin/AllocationsUsersImport.php

```diff
@@ -802,5 +815,55 @@ public function syncAGandCC($agBegin=0, $agEnd=1000) {
    */
   public function cleanObsoleteAllocations($indexStart, $indexStop) {
     $this->collectCronLog("Clean Obs: user count $indexStart, $indexStop ", 'i');
+    $obsoleteCount = 0;
+    try {
+      $portalNames = $this->getApiPortalUserNames();
+      $this->collectCronLog("Clean Obs: portal count " . count($portalNames), 'i');
+      if ($portalNames) {
+
+        // Make destination list of emails of users with administrator role.
+        $userIds = \Drupal::entityQuery('user')
+          ->condition('status', 1)
+          ->execute();
+        $userCount = count($userIds);
+        $this->collectCronLog("Clean Obs: user count $userCount ", 'i');
+
+        $index = 0;
+        foreach ($userIds as $userId) {
+
+          $index++;
+          if ($index < $indexStart) {
+            continue;
+          }
+          if ($index > $indexStop) {
+            break;
+          }
+
+          $user = User::load($userId);
+
+          $userCiderArray = $user->get('field_cider_resources')->getValue();
+          // $this->collectCronLog($index . '. ' . $userId . ' ' . $user->getAccountName() . ' - ' . count($userCiderArray), 'i');
+          // If ciders are listed on user's account, see if they are on the allocaions api list
+          // of active users.
+          if (count($userCiderArray) > 0) {
+
+            // Username without @access-ci.org is the portal name.
+            $accountSub = str_replace("@access-ci.org", "", $user->getAccountName());
+            if (strlen($accountSub) < strlen($user->getAccountName())) {
```

Comment:
I'm curious about the the purpose of this comparison? Maybe add a comment explaining?